run-cgprna/run_cgprna/map.py

```diff
@@ -0,0 +1,135 @@
+        '--species': 'unspecified_species',
+        '--reference-build': 'unspecified_ref_build',
+        '--gene-build': 'unspecified_gene_build'
+    }
+
+    # only use temp_dir when needed to extract reference files
+    temp_dir = os.path.join(os.path.abspath(args.out_dir), 'cgpRna_map_temp')
+    clean_temp = 0
+
+    other_options = []
+    if args.rg_id_tag:
+        other_options.append('-lane-id "%s"' % args.rg_id_tag)
+    if args.lb_tag:
+        other_options.append('-library "%s"' % args.lb_tag)
+    if args.ds_tag:
+        other_options.append('-ds-tag "%s"' % args.ds_tag)
+    if args.pl_tag:
+        other_options.append('-machine-type "%s"' % args.pl_tag)
+    if args.pu_tag:
+        other_options.append('-npg-run "%s"' % args.pu_tag)
+
+    # prepare the output dir
+    mkdir(args.out_dir)
+
+    # prepare the reference
+    reference_data_root = os.path.abspath(args.ref)
+    # Anything not a file will be treated as a reference root folder
+    if os.path.isfile(reference_data_root):
+        # If a pre-built ref bundle tar file is supplied
+        if re.match(r'.*\.tar\.gz$', os.path.basename(reference_data_root)):
+            mkdir(temp_dir)
+            clean_temp = 1
+
+            # use the tar file name as the ref_root_dir_name, so that in BAM header, people can tell which bundle file was used
+            ref_root_dir_name = re.match(r'(.*)\.tar\.gz$', os.path.basename(reference_data_root)).group(1)
+            reference_data_root=os.path.join(temp_dir, ref_root_dir_name)
+            
+            # set ref related values for star_mapping.pl
+            for key,value in ref_related_args.items():
+                # GTF file name will be the same as in the bundle
+                if key!= '--gene-build-gtf-name':
+                    if value is None:
+                        ref_related_args[key] = ref_related_defaults[key]
+                else:
+                    if value is not None:
+                        print('Warning: provided "--gene-build-gtf-name" will be overwritten by the GTF file name in the reference bundle.')
+
+            # make the folder structure
+            bundle_decompress_path = os.path.join(temp_dir, ref_root_dir_name, ref_related_args['--species'], ref_related_args['--reference-build'], 'star')
+            final_gtf_folder = os.path.join(bundle_decompress_path, ref_related_args['--gene-build'])
+            mkdir(final_gtf_folder)
+
+            # dump reference bundle
+            untar(args.ref, bundle_decompress_path)
+
+            # find the GTF file
+            gtfs = find('*.gtf', bundle_decompress_path)
+            if len(gtfs) == 1:
+                ref_related_args['--gene-build-gtf-name'] = os.path.basename(gtfs[0])
+                # link the file to final_gtf_folder
+                os.symlink(
+                    gtfs[0],
+                    os.path.join(final_gtf_folder, ref_related_args['--gene-build-gtf-name'])
+                )
+            else:
```

Comment:
I would suggest [guard clauses](https://www.refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html) instead of `else` statements when returning or exiting, to improve readability.